File: delete-node-in-a-linked-list/delete-node-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, node):
        '''
        Deletes a middle node in a linked list that is not a tail node
        Returns the updated node
        '''
        node.val = node.next.val
        node.next = node.next.next

File: delete-node-in-a-linked-list/delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution 
{
    public void deleteNode(ListNode node) 
    {
        node.val = node.next.val; // Swap the values
        node.next = node.next.next; // Delete the next node
    }
}